src/H5I.c

```diff
@@ -756,6 +802,7 @@ H5Iregister(H5I_type_t type, const void *object)
  *
  *-------------------------------------------------------------------------
  */
+H5_GCC_DIAG_OFF("implicit-fallthrough")
```

Comment:
can you explain why that needs to be silenced?

Reply:
uthash is implemented as a bunch of macros in a header file and some of those macros use fall-through. To fix this, I have to turn off the warnings in the H5I calls that use uthash. I can't just drop the 'diagnostic off' warnings in the header file since the macros just get pasted in and older versions of gcc require diagnostic switches to be at the function level.

Reply:
We do this in HDF5 where we use implicit fall-through in things like Duff's device.

Reply:
uthash.h seems to use /* FALLTHROUGH */ comments to mark purposeful fallthrough, why isn't GCC respecting those?

Reply:
We set -Wimplicit-fallthrough=5, which ignores comments and attributes, so the only thing you can do is turn it off. -Wextra sets -Wimplicit-fallthrough=3, which is probably suitable for our purposes (the integer specifies which comments match in the regexp . I'm not going to do that here, though.

I'll set up a separate PR that removes -Wimplicit-fallthrough and the now redundant diagnostic on/off statements.

Reply:
This was fixed in develop and merged here. We no longer ignore fall-through comments in gcc.